File: frontend/src/app/models/empleado.ts
/** Datos basicos del empleado para listados y selects */
export interface Empleado {
  id: number;
  nombre: string;
  apellido1: string;
  apellido2: string;
}

/** Empleado resumido para el listado de admin (incluye departamento) */
export interface EmpleadoResumen {
  id: number;
  nombre: string;
  apellido1: string;
  apellido2: string;
  departamento: string;
}

/** Empleado completo para la ficha de detalle */
export interface EmpleadoDetalle {
  id: number;
  nombre: string;
  apellido1: string;
  apellido2: string;
  nif: string;
  numeroSs: string;
  direccion: string;
  provincia: string;
  localidad: string;
  codigoPostal: number;
  fechaNacimiento: string;
  iban: string;
  telefono: string;
  departamentoId: number;
  departamento: string;
}

/** Datos editables del empleado para el PUT */
export interface EmpleadoEdicion {
  nombre: string;
  apellido1: string;
  apellido2: string;
  nif: string;
  numeroSs: string;
  direccion: string;
  provincia: string;
  localidad: string;
  codigoPostal: number;
  fechaNacimiento: string;
  iban: string;
  telefono: string;
  departamentoId: number;
}

/** Datos para crear un empleado nuevo (POST /empleados) */
export interface EmpleadoCreacion {
  email: string;
  password: string;
  privilegios: string[];
  nombre: string;
  apellido1: string;
  apellido2: string;
  nif: string;
  numeroSs: string;
  direccion: string;
  provincia: string;
  localidad: string;
  codigoPostal: number;
  fechaNacimiento: string;
  iban: string;
  telefono: string;
  departamentoId: number;
}

/** Departamento para el select */
export interface Departamento {
  id: number;
  nombre: string;
}

File: backend/src/main/java/apirest/workzen/modelo/entities/Empleado.java
package apirest.workzen.modelo.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonBackReference;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
@Table(name = "empleados")
public class Empleado implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @OneToOne
    @JoinColumn(name = "usuario_id")
    @JsonBackReference
    private Usuario usuario;

    @ManyToOne
    @JoinColumn(name = "departamento_id", nullable = false)
    private Departamento departamento;

    private String nombre;
    private String apellido1;
    private String apellido2;
    private String nif;

    @Column(name = "numero_ss")
    private String numeroSS;

    private String direccion;
    private String provincia;
    private String localidad;

    @Column(name = "codigo_postal")
    private Integer codigoPostal;

    @Column(name = "fecha_nacimiento")
    private LocalDate fechaNacimiento;

    private String iban;
    private String telefono;

    /** Dias de vacaciones totales asignados al empleado para el ano actual. */
    @Column(name = "dias_vacaciones")
    private int diasVacaciones;
}

























